flat_mutation_reader.hh

```diff
@@ -360,6 +360,19 @@ public:
         return _impl->consume(std::move(consumer), timeout);
     }
 
+    class filter {
+    private:
+        std::function<bool (const dht::decorated_key&)> partition_filter;
+    public:
+        filter(std::function<bool (const dht::decorated_key&)>&& pf = [] (const dht::decorated_key&) { return true; })
```

Comment:
Maybe it should also have a forwarding constructor:
```
template <typename Functor>
filter(Functor&& f) : partition_filter(std::forward<Functor>(f)) { }
```
This way existing code would not have to be changed, the lambdas they pass in would implicitly be converted to this filter.

Reply:
Good idea, thanks. Will add it.